pkg/updatechecker/updatechecker.go

```diff
@@ -210,16 +210,18 @@ func CheckForUpdates(opts CheckForUpdatesOpts) (ucr *UpdateCheckResponse, finalE
 	}
 
 	finishedChan := make(chan error)
-	if opts.Wait {
-		defer close(finishedChan)
-	}
+	defer func() {
+		if opts.Wait || finalError != nil {
+			defer close(finishedChan)
```

Comment:
shouldnt this just be `close(finishedChan)` since its already within a defer statement?  I dont think the extra defer here is necessary, unless i am missing somehting.

Reply:
this spins up a go routine if `Wait` is true.  The channel needs to stay open until that terminates.